printing/printer.ml

```diff
@@ -342,15 +349,55 @@ let pr_ne_context_of header env sigma =
     List.is_empty (Environ.named_context env)  then (mt ())
   else let penv = pr_context_unlimited env sigma in (header ++ penv ++ fnl ())
 
+(* Heuristic for horizontalizing hypothesis:
+   Detecting variable which type is a simple id or of the form (t x y ...)
+   where t is a product or only sorts (typically [Type -> Type -> ...]
+   and not [nat -> nat -> ...] ).
+   + Special case for non-Prop dependent terms. *)
+let rec should_compact env sigma typ =
+  get_compact_context() &&
+    match kind_of_term typ with
+    | Rel _ | Var _ | Sort _ | Const _ | Ind _ -> true
+    | App (c,args) ->
+       let _,type_of_c = Typing.type_of env sigma (EConstr.of_constr c) in
+       let _,type_of_typ = Typing.type_of env sigma (EConstr.of_constr typ) in
+       not (is_Prop (EConstr.to_constr sigma type_of_typ))
+       && (* These two more tests detect rare cases of non-Prop-sorted
+             dependent hypothesis:  *)
+       let lnamedtyp , _ = EConstr.decompose_prod sigma type_of_c in
+       (* c has a non dependent type *)
```

Comment:
I am not completely familiar with the vocabulary but to me a dependent type is one that depends on a term (i.e. on a term that is not also a type). That is: List is not a dependent Type, Vector is a dependent type. This test tries to discriminate between these two. But indeed the type of Vector is not a "dependent product", it is still a "dependent type" to me.
Is it a miswording?